src/lib/handlePageRequest.ts

```diff
@@ -98,9 +99,11 @@ const handlePostRequest = async (
 
   if (formIsValid) {
     let destination = onSuccessfulFormPostCallback(context);
-    destination = formatUrlQueryParams(destination, {
-      useIndex: context.useIndex,
-    });
+    if (!context.req.headers.referer.includes(PageURLs.signUpOrSignIn)) {
+      destination = formatUrlQueryParams(destination, {
+        useIndex: context.useIndex,
+      });
+    }
```

Comment:
> quick and easy but not very nice if statement

<img src="https://media.giphy.com/media/JqDeI2yjpSRgdh35oe/giphy.gif" width=200></img>

Reply:
I think you're very right to be unsure about this... and it's not your fault because the whole thing is a collection of not very nice if statements that are difficult to unpick... 😬

I'm also hesitant to put another if statement inside `handlePageRequest` as that would be kicking the can down the road!!!  Going to do some thinking about this - I think we should perhaps hold off on this PR until we've got the plumbing ready?

Reply:
@matthew-a-carr @tm1303 discussed this today.  We're going to _tolerate_ the `?useIndex=0` param on the signup etc. pages as that's less bad than adding another special case to `handlePageRequest` 👍